esrally/utils/console.py

```diff
@@ -145,12 +147,15 @@ def progress(width=90):
 class CmdLineProgressReporter:
     """
     CmdLineProgressReporter supports displaying an updating progress indication together with an information message.
+
+    :param custom_print: allow use of a different print method to assist with patching in unittests
     """
 
-    def __init__(self, width, plain_output=False):
+    def __init__(self, width, plain_output=False, printer=print):
         self._width = width
         self._first_print = True
         self._plain_output = plain_output
+        self._custom_print = printer
```

Comment:
Nit: How about we also call the field `_printer`?

Reply:
Fixed in https://github.com/elastic/rally/pull/700/commits/c45a2b92d1e800b0ee7b46e7bd27341048ee56d3